plugins/wandb/src/flyteplugins/wandb/tracking.py

```diff
@@ -0,0 +1,77 @@
+import os
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from flyte.extend import AsyncFunctionTaskTemplate, TaskPluginRegistry
+
+
+@dataclass(kw_only=True)
+class WandbConfig(object):
+    """
+    Wandb configuration for tracking experiments.
+
+    Attributes:
+        project (str): the name of the project where you're sending the new run. (Required)
+        wandb_api_key (Optional[str]): API key for Weights & Biases.
+        api_host (str, optional): URL to your API Host, The default is "https://api.wandb.ai".
+        entity (str, optional):
+        id (str, optional): A unique id for this wandb run.
+        **init_kwargs (dict): The rest of the arguments are passed directly to `wandb.init`.
+        Please see [the `wandb.init` docs](https://docs.wandb.ai/ref/python/init) for details.
+    """
+
+    project: str
+    wandb_api_key: Optional[str] = None
+    api_host: Optional[str] = None
+    entity: Optional[str] = None
+    id: Optional[str] = None
+    init_kwargs: Optional[Dict[str, Any]] = None
+
+
+@dataclass(kw_only=True)
+class WandbTracker(AsyncFunctionTaskTemplate):
+    """
+    Actual Plugin that track
+    Attributes:
+        plugin_config (Wandb): Configuration for Weights & Biases tracking.
+        task_type (str): Type of the task, It should be "wandb".
+    """
+
+    plugin_config: WandbConfig
+    task_type: str = "wandb"
+
+    async def execute(self, *args, **kwargs) -> Any:
```

Comment:
I do not think it should be a plugin, example, folks can use wandb with ray, spark, distributed training etc